pdf/pdfcanvas.cpp

```diff
@@ -164,12 +164,14 @@ void PDFCanvas::setDocument(PDFDocument* doc)
     }
 }
 
-qreal PDFCanvas::pagePosition(int index) const
+qreal PDFCanvas::pagePosition(int index, qreal offset) const
 {
     if( d->pages.count() == 0 )
         return 0.f;
 
-    return d->pages.value( index ).rect.y();
+    const QRectF& rect = d->pages.value( index ).rect;
+    qreal minOffset = 0., maxOffset = 1.;
+    return rect.y() + rect.height() * qBound(minOffset, offset, maxOffset);
```

Comment:
I agree to change offset that's not sounding for what it is. Maybe something like "qreal inPage".

For the rest, I may have not understood your point. For me, this value is mandatory in the function because every single page height is not known from QML side and because of spacing between pages, it is a bit cumbersome to use pagePosition(i+1) - pagePosition(i). ContentX is different because it refers to the width which is  known from the QML side.